src/locator/al-locator.types.ts

```diff
@@ -242,46 +285,56 @@ export class AlLocatorMatrix
     }
 
     /**
-     *  Recursively resolves the URI of a service node.
+     *  Nested nodes (e.g., an application living inside another application) are official dead, making this method
+     *  @deprecated.
      */
     public resolveNodeURI( node:AlLocationDescriptor, context?:AlLocationContext ):string {
-        if ( node._fullURI ) {
-            return node._fullURI;
-        }
-        let uri = '';
-        if ( node.parentId ) {
-            const parentNode = this.getNode( node.parentId, context );
-            if(parentNode) {
-                uri += this.resolveNodeURI( parentNode, context );
-            }
-        }
-        if ( node.uri ) {
-            uri += node.uri;
-            if ( ! node.parentId ) {
-                //  For historical reasons, some nodes (like auth0) are represented without protocols (e.g., alertlogic-integration.auth0.com instead of https://alertlogic-integration.auth0.com).
-                //  For the purposes of resolving functional links, detect these protocolless domains and add the default https:// protocol to them.
-                if ( uri.indexOf("http") !== 0 ) {
-                    uri = "https://" + uri;
-                }
-            }
-        }
-        node._fullURI = uri;
-        return uri;
+        console.warn("Deprecation warning: please do not use resolveNodeURI directly; just use the location's 'uri' property." );
+        return node.uri;
     }
 
     /**
-     *  Updates the service matrix model with a set of service node descriptors.  Optionally
+     *  Updates the locator matrix model with a set of service node descriptors.  Optionally
      *  calculates which node is the acting node based on its URI.
      *
      *  @param {Array} nodes A list of service node descriptors.
      */
     public setLocations( nodes:AlLocationDescriptor[] ) {
+        nodes.forEach( node => {
+            if ( node.environment && node.residency ) {
```

Comment:
I looked at this block of if statements and was slightly confused if the fall thru was intentional, noticed it was, then ay reversing the order made it a bit more clear and DRYd up the keys a bit.
```typescript
            let key = `${node.locTypeId}`;
            this.nodeDictionary[`${key}-*-*`] = node;
            if ( node.environment ) {
            key += -${node.environment}
                this.nodeDictionary[`${key}-*`] = node;
                if ( node.residency ) {
                key += -${node.residency};
                    this.nodeDictionary[key] = node;
                    if ( node.insightLocationId ) {
                        this.nodeDictionary[`${key}-${node.insightLocationId}`] = node;
                    }
                }
            }
```

Reply:
Nice.